spyder/api/plugins/new_api.py

```diff
@@ -1,228 +1,398 @@
-    #     ('section-name', {'option-1': 'some-value',
-    #                       'option-2': True,}),
-    #     ('another-section-name', {'option-3': 'some-other-value',
-    #                               'option-4': [1, 2, 3],}),
-    # ]
-    CONF_DEFAULTS = None
-
-    # Define configuration version if using a separate file
-    #
-    # IMPORTANT NOTES:
-    # 1. If you want to *change* the default value of a current option, you
-    #    need to do a MINOR update in config version, e.g. from 3.0.0 to 3.1.0
-    # 2. If you want to *remove* options that are no longer needed or if you
-    #    want to *rename* options, then you need to do a MAJOR update in
-    #    version, e.g. from 3.0.0 to 4.0.0
-    # 3. You don't need to touch this value if you're just adding a new option
-    CONF_VERSION = None
-
-    # Widget to be used as entry in Spyder Preferences dialog.
-    CONF_WIDGET_CLASS = None
-
-    # Some plugins may add configuration options for other plugins.
-    # Example:
-    # ADDITIONAL_CONF_OPTIONS = {'section': <new value to add>}
-    ADDITIONAL_CONF_OPTIONS = None
-
-    # Define additional configurable options (via a tab) to
-    # another's plugin configuration page. All configuration tabs should
-    # inherit from `SpyderPreferencesTab`.
-    # Example:
-    # ADDITIONAL_CONF_TABS = {'plugin_name': [<SpyderPreferencesTab classes>]}
-    ADDITIONAL_CONF_TABS = None
-
-    # Define custom layout classes that the plugin wantes to be registered.
-    # THe custom classes should extend from
-    #       `spyder.pluginsl.layout.api::BaseGridLayoutType`
-    CUSTOM_LAYOUTS = []
-
-    # Path for images relative to the plugin path
-    # A Python package can include one or several Spyder plugins. In this case
-    # the package may be using images from a global folder outside the plugin
-    # folder
-    IMG_PATH = None
-
-    # Control the font size relative to the global fonts defined in Spyder
-    MONOSPACE_FONT_SIZE_DELTA = 0
-    INTERFACE_FONT_SIZE_DELTA = 0
-
-    # Define context to store actions, toolbars, toolbuttons and menus.
-    CONTEXT_NAME = None
-
-    # Define if a plugin can be disabled in preferences.
-    # If False, the plugin is considered "core" and therefore it cannot be
-    # disabled. Default: True
-    CAN_BE_DISABLED = True
-
-    # Qt Web Widgets may be a heavy dependency for many packagers
-    # (e.g. conda-forge)
-    # We thus ask plugins to declare whether or not they need
-    # web widgets to enhance the distribution of Spyder to users
-    # https://github.com/spyder-ide/spyder/pull/22196#issuecomment-2189377043
-    REQUIRE_WEB_WIDGETS = False
+    # -----------------------------------------------------------------------
+
+    REQUIRES: list[str] = []
+    """
+    List of names of required dependencies for this plugin.
+
+    If a plugin uses a widget from another plugin, that other plugin must
+    be declared as a required dependency.
+
+    .. note::
```

Comment:
```suggestion
    List of names of required dependencies for this plugin.

    .. note::
```

Remove whitespace correctly.